[Build System: update-checkout] avoid multiprocessing when n is 1

hnrklssn opened the pull request:
There's no need to fiddle with locks etc. when there's no actual parallelism going on.

hnrklssn committed:
[Build System: update-checkout] avoid multiprocessing when n is 1

There's no need to fiddle with locks etc. when there's no actual
parallelism going on.

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented:
For context, this helped me work around an issue where I kept getting stuck waiting on a lock (for reasons I don't know).

MaxDesiatov reviewed (commented):
Seems fine, but lack of testing worries me. I'm not proficient enough with Python to suggest a concrete testing strategy though.

hnrklssn commented:
> Seems fine, but lack of testing worries me. I'm not proficient enough with Python to suggest a concrete testing strategy though.

Does update-checkout have tests?

MaxDesiatov commented:
It does seem to have some tests here https://github.com/swiftlang/swift/tree/main/utils/update_checkout/tests

hnrklssn committed:
[update-checkout] test single threaded checkout

hnrklssn committed:
[update-checkout] don't sign commits in mock repo

The mock repo is configured with the no-reply@swift.org email address.
If the user has globally configured git to sign commits this won't match
the signature, preventing the tests from running properly.

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented:
@MaxDesiatov Sorry about the delay. Added some single threaded versions of the existing tests. Let me know if you have any other comments.

MaxDesiatov commented:
@swift-ci smoke test linux

MaxDesiatov reviewed (approved):
Thanks!

AnthonyLatsis reviewed (commented):
LGTM, but please wait for a review from <s>Eric</s> Evan too. Feel free to [ping them](https://github.com/swiftlang/swift/blob/cc145482de3b9b9a271e7e7cba5aabebdbe8ef13/docs/HowToGuides/FirstPullRequest.md#i-didnt-get-a-response-from-someone-what-should-i-do) if the PR stalls.

hnrklssn commented:
@swift-ci smoke test linux
